src/components/AddTask.js

```diff
@@ -218,13 +244,43 @@ const AddTask = () => {
-                                    <li>{subtask}</li>
-                                    <div className="flex">
-                                        <img src={'./assets/icon/add-task/edit-icon.png'} alt={"plus icon"}/>
-                                        <img src={'./assets/icon/add-task/dr-icon.png'} alt={"plus icon"}/>
-                                        <img onClick={() => {deleteSubtask(index)}} src={'./assets/icon/add-task/trash-icon.png'} alt={"plus icon"}/>
-                                    </div>
+                                    {
+                                        subtasks[index].editOpen &&
+                                        <div className="flex items-center w-full h-12 border-b border-gray-300">
+                                            <input value={editSubtaskValue} onChange={(e) => setEditSubtaskValue(e.target.value)}/>
+                                            <div className="flex">
+                                                <img onClick={() => {
+                                                    cancelEdit(index)
+                                                }} src={'./assets/icon/add-task/x-cross.blue.svg'}
+                                                     alt={"X icon in blue"}/>
+                                                <img src={'./assets/icon/add-task/dr-icon.png'} alt={"plus icon"}/>
+                                                <img onClick={() => {
+                                                    acceptEdit(index, editSubtaskValue)
+                                                }} src={'./assets/icon/add-task/checkmark-blue.svg'}
+                                                     alt={"checkmark icon in blue"}/>
+                                            </div>
+                                        </div>
+                                    }
+
+
+                                    {
+                                        !subtasks[index].editOpen &&
+                                        <>
+                                            <li>{subtask.subtask}</li>
+                                            <div className="flex">
+                                                <img onClick={() => {
+                                                    editSubtask(index)
+                                                }} src={'./assets/icon/add-task/edit-icon.png'} alt={"plus icon"}/>
+                                                <img src={'./assets/icon/add-task/dr-icon.png'} alt={"plus icon"}/>
+                                                <img onClick={() => {
+                                                    deleteSubtask(index)
+                                                }} src={'./assets/icon/add-task/trash-icon.png'} alt={"plus icon"}/>
+                                            </div>
+                                        </>
+
+                                    }
                                 </div>
```

Comment:
This whole part is good! The only problem is that it can get quite long.
That <div> can be inside its own component, to allow you to switch between display and editing modes.